market/mysterium/response.go

```diff
@@ -22,24 +22,24 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net/http"
-)
 
-type errorResponse struct {
-	Error string `json:"error"`
-}
+	"github.com/pkg/errors"
+)
 
 // ParseResponseError checks the respose for correctness and return error if it is invalid
-func ParseResponseError(response *http.Response) error {
+func ParseResponseError(response *http.Response) (err error) {
 	if response.StatusCode < 200 || response.StatusCode >= 300 {
-		var error errorResponse
+		var errorMsg []byte
 
-		err := ParseResponseJSON(response, &error)
-		if err != nil {
-			return err
+		if response.Body != nil {
+			errorMsg, err = ioutil.ReadAll(response.Body)
+			if err != nil {
+				return errors.Wrap(err, "server response invalid")
```

Comment:
I guess my previous comment was a bit confusing :)
just do fmt.Errorf("server response invalid. Body reading error %s", err) to add more information. That's it

Reply:
But `errors.Wrap` [does](https://godoc.org/github.com/pkg/errors#Wrap) the same. But as an extra option, it allows getting more details if you need id for debugging if you want it. 
By default, it will be the same error as with `fmt.Errorf`.

And some people [recommends](https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully) to annotate errors with a `errors.Wrap`.

Reply:
In that case we must cease to use fmt.Errorf and use that error package. Sounds like one of team decision and guideline for me

Reply:
It would be good to have an actual stack trace, @soffokl , but for this we would probably need to change how we print errors. In general i agree that we need to wrap error causes.